### PR TITLE
Service borg tray no longer drap inventory version of items (stack count number)

### LINKED ISSUE #23825: Service cyborg tray drops inventory version of items
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

515.1619

### Issue Description

When the service cyborg uses their tray on a table, it drops the inventory version of the item, complete with stack count, hover highlight, and tooltip.

### What did you expect to happen?

The items should be separated and placed on the table.

### What happened instead?

The inventory version is placed on the table.

### Why is this bad/What are the consequences?

It definitely looks weird, and I'm not even sure if they can be separated by anyone else.

### Steps to reproduce the issue.

Become a service cyborg
Dispense two glasses with your fabricator
Pick the glasses up with a tray
Place the glasses on a table
Stack version of item is placed

### When did the problem start happening?

Unknown

### Extra information

_No response_

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by PIerreLouisH**
What Does This PR Do
Service borg tray no longer drap inventory version of items.
Fixes #23825

## Why It's Good For The Game
Bug fix gud

## Images of changes
image

## Changelog
:cl:
fix: Service borg tray no longer drap inventory version of items (stack count number)
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/storage/bags.dm [hunk 1]
```diff
@@ -479,15 +479,17 @@
 		var/dropped_something = FALSE
 
 		for(var/obj/item/I in contents)
-			I.loc = dropspot
+			var/obj/item/newItem = new I.type(dropspot)
+			// Set the properties of the new item here, e.g., stack count, hover highlight, tooltip
 			contents.Remove(I)
+			qdel(I) // Delete the old item
```
Comment (Contrabang) [inline code/game/objects/items/weapons/storage/bags.dm:485]: Deleting an item and replacing it with the same type is a terrible solution for this. This should use something like `remove_from_storage` to properly remove the maptext.

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/storage/bags.dm [hunk 2]
```diff
@@ -479,15 +479,17 @@
 		var/dropped_something = FALSE
 
 		for(var/obj/item/I in contents)
-			I.loc = dropspot
+			var/obj/item/newItem = new I.type(dropspot)
```
Comment (Contrabang) [inline code/game/objects/items/weapons/storage/bags.dm:482]:
```suggestion
			remove_from_storage(I)
```

File: code/game/objects/items/weapons/storage/bags.dm [hunk 3]
```diff
@@ -479,15 +479,17 @@
...
+			// Set the properties of the new item here, e.g., stack count, hover highlight, tooltip
 			contents.Remove(I)
+			remove_from_storage()
```
Comment (Contrabang) [inline code/game/objects/items/weapons/storage/bags.dm:485]:
```suggestion
```

File: code/game/objects/items/weapons/storage/bags.dm [hunk 4]
```diff
@@ -479,15 +479,17 @@
...
 			dropped_something = TRUE
 			if(!found_table && isturf(dropspot))
 				// if no table, presume that the person just shittily dropped the tray on the ground and made a mess everywhere!
 				spawn()
 					for(var/i = 1, i <= rand(1,2), i++)
-						if(I)
-							step(I, pick(NORTH,SOUTH,EAST,WEST))
+						if(newItem)
+							step(newItem, pick(NORTH,SOUTH,EAST,WEST))
```
Comment (Contrabang) [inline code/game/objects/items/weapons/storage/bags.dm:492]:
```suggestion
						if(I)
							step(I, pick(NORTH,SOUTH,EAST,WEST))
```

**Review by Contrabang — CHANGES_REQUESTED** [review]
These should be the correct changes, instead of making duplicate items, try this. Please make sure to test this also

**Comment by PIerreLouisH** [comment]
> These should be the correct changes, instead of making duplicate items, try this. Please make sure to test this also

Thank you for your time. The code does work without error technically, but the sodas cans will be put on an odd location. Will fix soon with a new commit.

**Comment by S34NW** [comment]
PR is stale, please continue working on it or address any comments within 7 days or we will close the PR.

**Review by Contrabang — APPROVED** [review]
LGTM

**Pull request merged**